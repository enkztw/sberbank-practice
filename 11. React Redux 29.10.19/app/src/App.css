#root {
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 100vw;
  min-height: 100vh;
}

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 80vw;
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}

th {
  padding: 10px;
}

tr.hovered {
  background-color: cyan;
  cursor: pointer;
}
