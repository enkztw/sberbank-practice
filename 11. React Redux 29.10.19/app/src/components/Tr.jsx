import React from "react";

const STATUS = {
  HOVERED: "hovered",
  NORMAL: "normal"
};

export default class extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      class: STATUS.NORMAL
    };
  }

  _onMouseEnter = () => {
    this.setState({ class: STATUS.HOVERED });
  }

  _onMouseLeave = () => {
    this.setState({ class: STATUS.NORMAL });
  }

  render() {
    return (
      <tr
        className={this.state.class}
        style={{ backgroundColor: this.props.color }}
        onMouseEnter={this._onMouseEnter}
        onMouseLeave={this._onMouseLeave}
      >
        {Object.values(this.props.data).map((item, index) => (
          <td key={index}>{item}</td>
        ))}
      </tr>
    );
  }
}
