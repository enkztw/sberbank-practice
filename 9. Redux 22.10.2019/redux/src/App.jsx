import React from "react";
import { connect } from "react-redux"
import { loadIssues } from './actions'
import { bindActionCreators } from "redux"

function App(props) {
  const handleClick = () => {
    props.loadIssues()
  }

  return (
    <React.Fragment>
      <button onClick={handleClick}>Load Issues</button>
      <ul>
      { props.issues.map((issue) => <li key={issue.title}>{issue.title}</li>)}
      </ul>
    </React.Fragment>
  );
}

export default connect(
  (state) => ({ counter: state.counter, issues: state.issues }),
  (dispatch) => bindActionCreators({loadIssues}, dispatch)
  )(App);
