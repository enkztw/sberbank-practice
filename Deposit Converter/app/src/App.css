html {
  font: 2vw "Roboto", sans-serif;
}

#root {
  display: flex;
  justify-content: center;
  align-items: center;

  height: 100vh;
  width: 50rem;
}

.board {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;

  width: 40rem;
  height: 15rem;
}

.bank {
  display: flex;
  flex-direction: column;
  align-items: center;

  box-sizing: border-box;
  width: 12rem;
  min-height: 15rem;

  padding: 1rem;


  box-shadow: 0.1rem 0.2rem 0.5rem rgba(0, 0, 0, 0.5);
  border-radius: 1rem;
  background-color: #ffffff;

  cursor: pointer;
  transition: 0.25s;
}

.bank:nth-of-type(2n) {
  transform: scale(1.1);
}

.bank:not(:nth-of-type(2n)):hover {
  transform: scale(1.05);

  transition: 0.25s;
}

.bank h1 {
  font-size: 1.4rem;
  font-weight: 400;

  margin: 0;
  border-bottom: 0.1rem solid #000000;
}

.bank img {
  width: 2rem;

  margin: 0.5rem 0 0.2rem;
}

.info {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  height: 100%;
}

.info h2 {
  font-size: 1.25rem;

  margin: 0;
}

.info .percent {
  font-size: 1.75rem;

  padding: 0.25rem 0;
}

.info .term {
  font-size: 0.65rem;
  color: #414142;
}