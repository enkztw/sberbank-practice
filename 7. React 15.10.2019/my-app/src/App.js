import React from 'react';
import './App.css';

const MyButton = (props) => {
  return <button onClick={props.handler} className='my-button'>{props.str}</button>
}

/**
 * Как высчитываются очки
 * 
 *.----------.------.-------.----------.
 *|          | Rock | Paper | Scissors |
 *:----------+------+-------+----------:
 *| Rock     |    0 |     0 |        1 |
 *:----------+------+-------+----------:
 *| Paper    |    1 |     0 |        0 |
 *:----------+------+-------+----------:
 *| Scissors |    0 |     1 |        0 |
 *'----------'------'-------'----------'
 * */

const truthTable = [
    [0,0,1],
    [1,0,0],
    [0,1,0]
]

const checkWinner = (playerChoice, computerChoice) => {
    const playerScore = truthTable[playerChoice][computerChoice];
    const computerScore = truthTable[computerChoice][playerChoice];
    if (playerScore > computerScore) {
        return 'Ты';
    } else if (computerScore > playerScore) {
        return 'Комп';
    } else if (playerScore === computerScore) {
        return 'Никто';
    }
}

const buttons = [{str: 'камень', type: 'rock', score: 0}, {str: 'бумага', type: 'paper', score: 1}, {str: 'ножницы', type: 'scissors', score: 2}]

export default class App extends React.Component {

    state = {
        compChoice: '',
        userChoice: '',
        winner: ''
    }
    
    getRandomNum = (min, max) => {
      min = Math.ceil(min)
      max = Math.floor(max)
      return Math.floor(Math.random() * (max - min + 1)) + min
  }

    onClickHandler = (score) => {
      const compChoice = this.getRandomNum(0, 2);
      const userChoice = score;
      this.setState({
        compChoice,
        userChoice,
      })
    }


  render() {
    return (
        <div className="App">
          <header className="App-header">
              <p>Камень, Ножницы, Бумага!</p>
          </header>
            <div>
                {buttons.map((button) => <MyButton {...button} handler={() => this.onClickHandler(button.score)} />)}
            </div>
            <div>
            <p>Комп выбрал: {this.state.compChoice && buttons[this.state.compChoice].str}</p>
            <p>А твой выбор: {this.state.userChoice && buttons[this.state.userChoice].str}</p>
            <p>Победитель: {this.state.userChoice && this.state.compChoice && checkWinner(this.state.userChoice, this.state.compChoice)}</p>
            </div>
        </div>
    );
  }
}
