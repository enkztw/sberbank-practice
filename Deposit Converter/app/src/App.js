import './css/App.css';
import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';

import Deposits from './components/Deposits';
import Calculator from './components/Calculator';


export default function Board() {
  return (
    <BrowserRouter>
      <Route exact path='/' component={Deposits} />
      <Route path='/calculator/:name' component={Calculator} />
    </BrowserRouter>
  )
}



