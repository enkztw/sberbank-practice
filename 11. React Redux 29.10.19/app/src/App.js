import React from 'react';
import './App.css';

import data from './data';
import Input from './components/Input'
import Tr from './components/Tr'

class App extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      data,
      searchValue: ''
    }
  }

  sortBy = (data, param) => [...data].sort((a, b) => b[param] - a[param])

  onChange = (e) => {
    this.setState({
      searchValue: e.target.value,
      data: this.sortBy(data.filter((item) => item.name.toLowerCase().startsWith(e.target.value.toLowerCase()), 'height'))
    })

    console.log(this.state.data)
  } 

  render() {
    return (
      <div className="App">
        <Input value={this.state.searchValue} onChange={this.onChange} />
        <table>
          <thead>
            <tr>
              {Object.keys(data[0]).map((key) => <th key={key}>{key}</th>)}
            </tr>
          </thead>
          <tbody>
            {this.sortBy(this.state.data, 'height').map((item, index) => <Tr key={item.name} data={item} color={index === 0 ? 'red' : ''} />)}
          </tbody>
        </table>
      </div>
    )
  }
}

export default App;
