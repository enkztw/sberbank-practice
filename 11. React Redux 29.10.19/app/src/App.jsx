import React from 'react';
import './App.css';
import { sortBy } from './utils';

import data from './data';
import Input from './components/Input'
import Tr from './components/Tr'

class App extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      data,
      searchValue: ''
    }
  }


  onChange = (e) => {
    this.setState({
      searchValue: e.target.value,
      data: sortBy(data.filter((item) => item.name.toLowerCase().startsWith(e.target.value.toLowerCase()), 'height'))
    })
  } 

  render() {
    return (
      <div className="App">
        <Input value={this.state.searchValue} onChange={this.onChange} />
        <table>
          <thead>
            <tr>
              {Object.keys(data[0]).map((key) => <th key={key}>{key}</th>)}
            </tr>
          </thead>
          <tbody>
            {sortBy(this.state.data, 'height').map((item, index) => <Tr key={item.name} data={item} color={index === 0 ? 'red' : ''} />)}
          </tbody>
        </table>
      </div>
    )
  }
}

export default App;
