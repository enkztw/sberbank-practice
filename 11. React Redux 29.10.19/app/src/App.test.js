import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Tr from './components/Tr';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});


test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Tr data={{0: 1, 1: 2, 2: 3}}></Tr>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});