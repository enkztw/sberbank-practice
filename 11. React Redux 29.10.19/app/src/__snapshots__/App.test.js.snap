// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<tr
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <td>
    1
  </td>
  <td>
    2
  </td>
  <td>
    3
  </td>
</tr>
`;

exports[`Link changes the class when hovered 2`] = `
<tr
  className="hovered"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <td>
    1
  </td>
  <td>
    2
  </td>
  <td>
    3
  </td>
</tr>
`;

exports[`Link changes the class when hovered 3`] = `
<tr
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <td>
    1
  </td>
  <td>
    2
  </td>
  <td>
    3
  </td>
</tr>
`;
